staging/src/k8s.io/apiserver/pkg/util/flowcontrol/filterconfig/apf_controller.go

```diff
@@ -0,0 +1,700 @@
+			}
+		}
+		var err error
+		plState.qsCompleter, err = qscOfPL(meal.cfgCtl.queueSetFactory, plState.queues, plName, &plState.config, meal.cfgCtl.requestWaitLimit)
+		if err != nil {
+			// This can not happen because qscOfPL already approved this config
+			panic(fmt.Sprintf("%s from name=%q spec=%#+v", err.Error(), plName, fcfmt.Fmt(plState.config)))
+		}
+		if plState.config.Limited != nil {
+			meal.shareSum += float64(plState.config.Limited.AssuredConcurrencyShares)
+		}
+		meal.haveExemptPL = meal.haveExemptPL || plName == fctypesv1a1.PriorityLevelConfigurationNameExempt
+		meal.haveCatchAllPL = meal.haveCatchAllPL || plName == fctypesv1a1.PriorityLevelConfigurationNameCatchAll
+		meal.newPLStates[plName] = plState
+	}
+}
+
+// For all the priority levels of the new config, divide up the
+// server's total concurrency limit among them and create/update their
+// QueueSets.
+func (meal *cfgMeal) finishQueueSetReconfigsLocked() {
+	for plName, plState := range meal.newPLStates {
+		if plState.config.Limited == nil {
+			klog.V(5).Infof("Using exempt priority level %q: quiescing=%v", plName, plState.quiescing)
+			continue
+		}
+
+		concurrencyLimit := int(math.Ceil(float64(meal.cfgCtl.serverConcurrencyLimit) * float64(plState.config.Limited.AssuredConcurrencyShares) / meal.shareSum))
+		metrics.UpdateSharedConcurrencyLimit(plName, concurrencyLimit)
+
+		if plState.queues == nil {
+			klog.V(5).Infof("Introducing queues for priority level %q: config=%#+v, concurrencyLimit=%d, quiescing=%v (shares=%v, shareSum=%v)", plName, fcfmt.Fmt(plState.config), concurrencyLimit, plState.quiescing, plState.config.Limited.AssuredConcurrencyShares, meal.shareSum)
+		} else {
+			klog.V(5).Infof("Retaining queues for priority level %q: config=%#+v, concurrencyLimit=%d, quiescing=%v, numPending=%d (shares=%v, shareSum=%v)", plName, fcfmt.Fmt(plState.config), concurrencyLimit, plState.quiescing, plState.numPending, plState.config.Limited.AssuredConcurrencyShares, meal.shareSum)
+		}
+		plState.queues = plState.qsCompleter.Complete(fq.DispatchingConfig{ConcurrencyLimit: concurrencyLimit})
+	}
+}
+
+// qscOfPL returns a pointer to an appropriate QueuingConfig or nil
+// if no limiting is called for.  Returns nil and an error if the given
+// object is malformed in a way that is a problem for this package.
+func qscOfPL(qsf fq.QueueSetFactory, queues fq.QueueSet, plName string, plSpec *fctypesv1a1.PriorityLevelConfigurationSpec, requestWaitLimit time.Duration) (fq.QueueSetCompleter, error) {
+	if (plSpec.Type == fctypesv1a1.PriorityLevelEnablementExempt) != (plSpec.Limited == nil) {
+		return nil, errors.New("broken union structure at the top")
+	}
+	if (plSpec.Type == fctypesv1a1.PriorityLevelEnablementExempt) != (plName == fctypesv1a1.PriorityLevelConfigurationNameExempt) {
+		// This package does not attempt to cope with a priority level dynamically switching between exempt and not.
+		return nil, errors.New("non-alignment between name and type")
```

Comment:
it wont be happening (this comment also applies to L539)

https://github.com/kubernetes/kubernetes/blob/29b09c7fb1a8d38ab9df4873553cc0b8d97aae95/pkg/apis/flowcontrol/validation/validation.go#L365-L367

Reply:
I got beaten up in previous reviews about long-distance dependencies.  There seems to be the idea of not relying on the validation because that is rather distant code.